crates/fluvio/src/spu.rs

```diff
@@ -78,7 +78,17 @@ impl SpuPool {
 
         debug!("connecting to spu: {}", spu.spec);
         let mut client_config = self.config.with_prefix_sni_domain(spu.key());
-        let spu_addr = spu.spec.public_endpoint.addr();
+
+        let spu_addr =
+            if self.config.use_spu_local_address && spu.spec.public_endpoint_local.is_some() {
```

Comment:
prob want something like: `if let Some(local) = spu.spec.public_endpoint`

Reply:
Yeah, this is kinda exactly what nick and I discussed when I wrote it.

Reply:
You can do a match with a guard, something like:

```rust
match spu.spec.public_endpoint_local {
    Some(endpoint) if self.config.use_spu_local_address => { ... }
    _ => { ... }
}
```